report/2.Analysis/ArduinoLanguagesCtrl.tex

```diff
@@ -75,18 +135,102 @@ \section{Existing languages}
 The VM includes many of the features that have made the Java VM (JVM) popular such as object-oriented programming, automatic dynamic memory allocation, and garbage collection.
 It comes packaged with native classes such as Object, System, and PrintStream.
 
-The downsides of using NanoVM instead of the Arduino language is, as stated on the official website of NanoVM\footnote{http://www.harbaum.org/till/nanovm/index.shtml} that the VM's interpreter doesn't perform as well as C compiled to AVR code, and it reserves some of the RAM used for applications to run the VM.
-NanoVM also has to be installed on the CPU's flash memory, which will overwrite the bootloader.
+The following is a short description of the basic Java syntax. Snippets \ref{lis:javaExample} and \ref{lis:javaPrinter} showcase the syntax.
+
+The program is started by calling the \texttt{main} function (snippet \ref{lis:javaExample}, line 5).
+
+All statements are finished with a semi-colon.
+
+In Java all variables and functions must be declared inside a class, and each class must be declared in a file for itself.
+
+\textit{Class fields} and methods are declared identically to how variables are declared in Arduino C++, except each field variable is prefixed with an optional access modifier (snippet \ref{lis:javaExample}, lines 3, 5).
+Variables can be made into class variables (called so because the variables belong to the class instead of instances of the class) with the \texttt{static} keyword before the name in the declaration.
+The same can be done to methods.
+
+\textit{Comments} are written exactly like in C and C++.
 
-A typical program written in Java is set up like this:
-\begin{lstlisting}[language=Java,label=lis:javaExample,caption=An example program written in Java,firstnumber=1]
+\begin{lstlisting}[language=Java,label=lis:javaExample,caption=An example program written in Java.,firstnumber=1]
 public class HelloWorld {
-	public static void main(String[] args) {
-		System.out.println("Hello World!");
+
+	private static Printer printer = new printer(); //Class variable
+	
+	public static void main(String[] args) { //Program start
+		printer.print("Hello World!");
+	}
+	
+}
+\end{lstlisting}
+
+\begin{lstlisting}[language=Java,label=lis:javaPrinter,caption=Printer class with one method written in Java.,firstnumber=1]
+public class Printer {
+	public void print(String str) { //Instance method
+		System.out.println(str);
 	}
 }
 \end{lstlisting}
 
+The downsides of using NanoVM instead of Arduino C++ is, as stated on the official website of NanoVM\footnote{http://www.harbaum.org/till/nanovm/index.shtml} that the VM's interpreter doesn't perform as well as C compiled to AVR code, and it reserves some of the RAM used for applications to run the VM.
```

Comment:
Starts an interposed phrase at "arduino c++ is, as.." but never ends. 
end it before "...that the VM's..."